media-gfx/curaengine/curaengine-4.6.1-r1.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake toolchain-funcs
+
+MY_PN="CuraEngine"
+
+DESCRIPTION="A 3D model slicing engine for 3D printing"
+HOMEPAGE="https://github.com/Ultimaker/CuraEngine"
+SRC_URI="https://github.com/Ultimaker/${MY_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="AGPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+arcus doc openmp test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	arcus? (
+		~dev-libs/libarcus-${PV}:*
+		dev-libs/protobuf:=
+	)
+	dev-libs/clipper
+	dev-libs/rapidjson
+	dev-libs/stb"
+BDEPEND="${RDEPEND}
+	|| ( sys-devel/gcc sys-devel/clang )
```

Comment:
This is not really a convincing way to check whether these are used, since both can be installed but something else used. Also gcc should be part of every installation. If musl is your concern, there are checks for that.

You can use *toolchain-funcs.eclass* to properly check whether gcc or clang is being used.